### PR TITLE
[WC-2023] Implement disabled state on DG2 export

### PR DESCRIPTION
**Pull request by samuelreichert**
### Pull request type

New feature (non-breaking change which adds functionality)

---

### Description
- Create a modal to disable any interaction with DataGrid2 while exporting
- Provide a progress bar with percentage of completion when total count is available
- Provide an indeterminate progress bar when no total count is available
- Provide a visual way to abort the exporting

### CODE REVIEW

File: packages/pluggableWidgets/datagrid-web/typings/global.d.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { DATAGRID_DATA_EXPORT, DataExporter, DataGridName } from "../src/feature
 declare global {
     interface Window {
         [DATAGRID_DATA_EXPORT]: Record<DataGridName, DataExporter>;
-        "DATAGRID_DATA_EXPORT": string
+        "DATAGRID_DATA_EXPORT": string;
+        __abort: () => void;
```
Comment (iobuhov) [inline packages/pluggableWidgets/datagrid-web/typings/global.d.ts:7]: I'm not sure that defining this function on window is needed. If you wan to make it available you should probably define it on some "scope" not directly on window

File: packages/pluggableWidgets/datagrid-web/src/components/ProgressModal.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+export const ProgressModal: FC<ProgressModalProps> = ({
+    failed,
+    onCancel,
+    open,
+    progress,
+    total
+}): ReactElement | null => {
+    const [isModalOpen, setIsModalOpen] = useState(false);
```
Comment (iobuhov) [inline packages/pluggableWidgets/datagrid-web/src/components/ProgressModal.tsx:23]: If you accept state (`props.open`) from props, then you don't need "internal" state. Always choose only one thing: ether - your component is "controlled" and then it don't have any state and always depends on props or - your component "uncontrolled" and it maintain it's own internal state.

File: packages/pluggableWidgets/datagrid-web/src/components/WidgetRoot.tsx [hunk 3]
```diff
@@ -21,6 +21,7 @@ export function WidgetRoot(props: WidgetRootProps): ReactElement {
                 "widget-datagrid-selection-method-click": selection && selectionMethod === "rowClick"
             })}
             {...rest}
+            style={{ position: "relative" }}
```
Comment (iobuhov) [inline packages/pluggableWidgets/datagrid-web/src/components/WidgetRoot.tsx:24]: Please, don't use inline styles. Inline styles very hard to override. If you need set position, use `_datagrids.scss`

**Review by iobuhov — CHANGES_REQUESTED** [review]

File: packages/pluggableWidgets/datagrid-web/src/Datagrid.tsx [hunk 4]
```diff
@@ -192,15 +193,20 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
                     </WidgetHeaderContext>
                 )
             }
-            hasMoreItems={props.datasource.hasMoreItems ?? false}
+            headerTitle={props.filterSectionTitle?.value}
             headerWrapperRenderer={useCallback((_columnIndex: number, header: ReactElement) => header, [])}
             id={id.current}
             numberOfItems={props.datasource.totalCount}
+            onDialogClose={abort}
```
Comment (iobuhov) [inline packages/pluggableWidgets/datagrid-web/src/Datagrid.tsx:200]: Let's rename `onDialogClose` to `onExportCancel`

**Review by iobuhov — CHANGES_REQUESTED** [review]

**Review by iobuhov — APPROVED** [review]

**Review by iobuhov — DISMISSED** [review]

**Review by iobuhov — DISMISSED** [review]

**Review by iobuhov — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**